test/schema-serialization.test.js

```diff
@@ -60,6 +60,100 @@ test('custom serializer options', t => {
   })
 })
 
+test('Different content types', t => {
+  t.plan(15)
+
+  const fastify = Fastify({
+    serializerOpts: {
+      rounding: 'ceil'
+    }
+  })
+
+  fastify.addSchema({
+    $id: 'test',
+    type: 'object',
+    properties: {
+      image: { type: 'string' }
+    }
+  })
+
+  fastify.get('/', {
+    schema: {
+      response: {
+        200: [
+          { content: 'application/json', type: { id: { type: 'number' }, name: { type: 'string' } } },
+          {
+            content: 'application/vnd.bla.v1+json',
+            type: {
+              type: 'array',
+              items: { $ref: 'test' }
+            }
+          }
+        ],
```

Comment:
the response should be an array of objects with two properties: 
- `content` referring to the content type 
- `type` referring to the actual schema. 

I'm feeling that `type` could be changed to something more meaningful, but I don't know what - in order for the people to not get overwhelmed between the expected `type` property and the actual types ( `id: { type: 'number' }` )